tests/contrib/langchain/conftest.py

```diff
@@ -1,73 +1,131 @@
+from http.server import BaseHTTPRequestHandler
+from http.server import HTTPServer
 import os
+import threading
 
-import mock
 import pytest
 
 from ddtrace.contrib.internal.langchain.patch import patch
 from ddtrace.contrib.internal.langchain.patch import unpatch
+from ddtrace.llmobs import LLMObs as llmobs_service
+from ddtrace.llmobs._writer import LLMObsSpanWriter
 from ddtrace.trace import Pin
 from tests.utils import DummyTracer
-from tests.utils import DummyWriter
-from tests.utils import override_config
 from tests.utils import override_env
 from tests.utils import override_global_config
 
 
+class TestLLMObsSpanWriter(LLMObsSpanWriter):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.events = []
+
+    def enqueue(self, event):
+        self.events.append(event)
+        super().enqueue(event)
+
+
 @pytest.fixture
-def ddtrace_config_langchain():
-    return {}
+def llmobs_env():
+    return {
+        "DD_API_KEY": "<default-not-a-real-key>",
+        "DD_LLMOBS_ML_APP": "unnamed-ml-app",
+    }
+
+
+class LLMObsServer(BaseHTTPRequestHandler):
+    """A mock server for the LLMObs backend used to capture the requests made by the client.
+
+    Python's HTTPRequestHandler is a bit weird and uses a class rather than an instance
+    for running an HTTP server so the requests are stored in a class variable and reset in the pytest fixture.
+    """
+
+    requests = []
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+
+    def do_POST(self) -> None:
+        content_length = int(self.headers["Content-Length"])
+        body = self.rfile.read(content_length).decode("utf-8")
+        self.requests.append({"path": self.path, "headers": dict(self.headers), "body": body})
+        self.send_response(200)
+        self.end_headers()
+        self.wfile.write(b"OK")
 
 
 @pytest.fixture
-def snapshot_tracer(langchain, mock_logs, mock_metrics):
-    pin = Pin.get_from(langchain)
-    yield pin.tracer
-    mock_logs.reset_mock()
-    mock_metrics.reset_mock()
+def _llmobs_backend():
+    LLMObsServer.requests = []
+    # Create and start the HTTP server
+    server = HTTPServer(("localhost", 0), LLMObsServer)
+    server_thread = threading.Thread(target=server.serve_forever)
+    server_thread.daemon = True
+    server_thread.start()
+
+    # Provide the server details to the test
+    server_address = f"http://{server.server_address[0]}:{server.server_address[1]}"
+
+    yield server_address, LLMObsServer.requests
+
+    # Stop the server after the test
+    server.shutdown()
+    server.server_close()
 
 
 @pytest.fixture
-def mock_tracer(langchain):
-    pin = Pin.get_from(langchain)
-    mock_tracer = DummyTracer(writer=DummyWriter(trace_flush_enabled=False))
-    pin._override(langchain, tracer=mock_tracer)
-    yield mock_tracer
+def llmobs_span_writer(_llmobs_backend):
```

Comment:
Is there a reason we are mocking/checking the overall request received by the llmobs backend instead of just checking the writer's enqueued events? Do we care about the end-to-end testing of the former approach?

Reply:
oh great question, i don't think there's a need to. i think when i was refactoring i was looking at our `llmobs` tests and might've just blindly copied 😅 will remove this for just testing the enqueued events!